### ISSUE TITLE
proftpd hides entries in root directory beginning with a DefaultRoot value (if mod_vroot is enabled)

### ISSUE THREAD

**Issue by pghmcfc** [proftpd/proftpd#1491]
This report is based on https://bugzilla.redhat.com/show_bug.cgi?id=2104972 which refers to proftpd-1.3.5e on RHEL7 but I am also able to reproduce it on proftpd-1.3.7e on Fedora 36, so long as `mod_vroot` is enabled.

An FTP client does not see entries in a root directory with names beginning with a `DefaultRoot` value, i.e.:
when `DefaultRoot` is `/store`, entries like `/store/store`, `/store/storeanything` are invisible for clients. When renamed to `/store/smth_else` they appear.

### What I Did

For example, I did a fresh install of proftpd 1.3.7e on Fedora 36 and changed the existing `DefaultRoot` value from the packaged default value of `~ !adm` to `store`. Then I created these directories:
```
# mkdir /store
# mkdir /store/storetest
# mkdir /store/store
# mkdir /store/storeagain
# mkdir /store/not-store
```
I started the FTP server and connected to it using my regular login:
```
$ ftp localhost
Trying ::1...
Connected to localhost (::1).
220 FTP Server ready.
Name (localhost:paul):
331 Password required for paul
Password:
230 User paul logged in
ftp> dir
229 Entering Extended Passive Mode (|||29121|)
150 Opening ASCII mode data connection for file list
drwxr-xr-x   2 root     root         4096 Jul 28 10:23 not-store
226 Transfer complete
ftp> bye
221 Goodbye.
```
As can be seen above, only the directory name that does not start with "store" is visible.
I then tried changing `VRootEngine` to `off` and tried again:
```
$ ftp localhost
Trying ::1...
Connected to localhost (::1).
220 FTP Server ready.
Name (localhost:paul):
331 Password required for paul
Password:
230 User paul logged in
Remote system type is UNIX.
Using binary mode to transfer files.
ftp> dir
229 Entering Extended Passive Mode (|||32407|)
150 Opening ASCII mode data connection for file list
drwxr-xr-x   2 root     root         4096 Jul 28 10:23 not-store
drwxr-xr-x   2 root     root         4096 Jul 28 10:23 store
drwxr-xr-x   2 root     root         4096 Jul 28 10:23 storeagain
drwxr-xr-x   2 root     root         4096 Jul 28 10:23 storetest
226 Transfer complete
ftp> quit
221 Goodbye.
```
The original bug report linked above contains a full config if you need it but this seems to be easy enough to reproduce without any special configuration.
My Fedora 36 install is using `mod_vroot` 0.9.9 and the original bug report will be with `mod_vroot` 0.9.2.

**Comment by Castaglia** [comment]
This is actually a bug in `mod_vroot`; I try to keep issues/tickets in their respective repos -- but here is fine, too.  This PR should hopefully address the issue: https://github.com/Castaglia/proftpd-mod_vroot/pull/37

@pghmcfc If you can confirm it works for you, I'll merge it, and publish a new version/release for `mod_vroot`.

**Comment by pghmcfc** [comment]
@Castaglia this looks to be working OK. I've tried backporting it to mod_vroot 0.9.2 for my RHEL7 build and that seems to work locally for me, so I'll ask the original reporter to give it a go.

**Comment by pghmcfc** [comment]
Original reporter seems to be happy with the fix.

**Comment by Castaglia** [comment]
> Original reporter seems to be happy with the fix.

Thanks!  I've merged that PR, and released [`mod_vroot-0.9.10`](https://github.com/Castaglia/proftpd-mod_vroot/releases/tag/v0.9.10).
